src/rlookup.c

```diff
@@ -70,7 +70,7 @@ RLookupKey *RLookup_GetKeyEx(RLookup *lookup, const char *name, size_t n, int fl
   int isNew = 0;
 
   for (RLookupKey *kk = lookup->head; kk; kk = kk->next) {
-    if (kk->name_len == n && !strncmp(kk->name, name, kk->name_len)) {
+    if (kk->name_len == n && (!strncmp(kk->name, name, kk->name_len) || !strcmp(kk->path, name))) {
```

Comment:
should path always override name?

Reply:
Can we avoid the second `strcmp` if `kk->name` and `kk->path` are the same pointer?

Reply:
@rafie Can you please elaborate?
@oshadmi :+1:

Reply:
Currently we give `name` precedence over `path`, meaning only if `name` does not match, we then try the `path`.
Do we test for ambiguities, since paths may still start with no dot or dollar, so they could match another field name.

Reply:
@oshadmi @rafie This function finds a single field against all fields in the lookup table.
There is no precedence here, if either matches, we return the `RLookupKey`.